### PR TITLE
Fix Go struct versioning, remove duplicate code, use embedding, reconcile differences

### PR DESCRIPTION
**Pull request by rob05c**
#### What does this PR do?

Fix TO/TC struct versioning, remove duplicate code, use embedding, reconcile differences

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] Other _________

#### What is the best way to verify this PR?

`go build`

#### Check all that apply

- [ ] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2377/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2378/
Test FAILed.

File: traffic_ops/client/deliveryservice.go [hunk 1]
```diff
@@ -56,25 +56,37 @@ func (to *Session) GetDeliveryServicesByServer(id int) ([]tc.DeliveryService, Re
...
+		return nil, reqInf, err
+	}
+
+	return data.Response, reqInf, nil
+}
+
 // DeliveryService gets the DeliveryService for the ID it's passed
 // Deprecated: use GetDeliveryService
-func (to *Session) DeliveryService(id string) (*tc.DeliveryService, error) {
+func (to *Session) DeliveryService(id string) (*tc.DeliveryServiceV13, error) {
```
Comment (moltzaum) [inline traffic_ops/client/deliveryservice.go]: Isn't this supposed to be just `DeliveryService`?

Comment (moltzaum) [reply traffic_ops/client/deliveryservice.go]: nvm, apparently that has changed already

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2379/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2381/
Test FAILed.

**Comment by dewrich** [comment]
swaggerdocs structs no longer compile:

```rob05c/to-fix-version ᐅ gt
     | swaggerdocs/v13/asns.go:3:8: cannot find package "github.com/apache/trafficcontrol/lib/go-tc/v13" in any of:
     | 	/Users/dricha209/projects/go/src/github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/vendor/github.com/apache/trafficcontrol/lib/go-tc/v13 (vendor tree)
     | 	/Users/dricha209/projects/go/src/github.com/apache/trafficcontrol/traffic_ops/vendor/github.com/apache/trafficcontrol/lib/go-tc/v13
     | 	/Users/dricha209/projects/go/src/github.com/apache/trafficcontrol/vendor/github.com/apache/trafficcontrol/lib/go-tc/v13
     | 	/usr/local/Cellar/go@1.9/1.9.7/libexec/src/github.com/apache/trafficcontrol/lib/go-tc/v13 (from $GOROOT)
     | 	/Users/dricha209/projects/go/src/github.com/apache/trafficcontrol/lib/go-tc/v13 (from $GOPATH)

```

**Comment by rob05c** [comment]
@dewrich Fixed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2386/
Test FAILed.

**Pull request merged**